src/modules/job-info/list.c

```diff
@@ -488,25 +488,47 @@ void list_id_cb (flux_t *h, flux_msg_handler_t *mh,
     json_decref (job);
 }
 
+int list_attrs_append (json_t *a, const char *attr)
+{
+    json_t *o = json_string (attr);
+    if (!o) {
+        errno = ENOMEM;
+        return -1;
+    }
+    if (json_array_append_new (a, o) < 0) {
+        json_decref (o);
+        errno = ENOMEM;
+        return -1;
+    }
+    return 0;
+}
+
 void list_attrs_cb (flux_t *h, flux_msg_handler_t *mh,
                     const flux_msg_t *msg, void *arg)
 {
-    if (flux_respond_pack (h, msg, "{s:[s,s,s,s,s,s,s,s,s,s,s,s,s,s]}",
-                           "attrs",
-                           "userid",
-                           "priority",
-                           "t_submit",
-                           "t_depend",
-                           "t_sched",
-                           "t_run",
-                           "t_cleanup",
-                           "t_inactive",
-                           "state",
-                           "name",
-                           "ntasks",
-                           "nnodes",
-                           "ranks",
-                           "success") < 0) {
+    json_t *a = NULL;
+
+    if (!(a = json_array ())) {
+        errno = ENOMEM;
+        goto error;
+    }
+    if (list_attrs_append (a, "userid") < 0
```

Comment:
Suggestion: use an array so code doesn't change if attr list changes, and array might be reusable in other contexts?

E.g.
```c
static const char *attr_table[] = { "priority",  ..., "success", NULL };
...
int i;
for (i = 0; attr_table[i] != NULL; i++) {
    if (list_attrs_append (a, attr_table[i]) < 0)
        /* error */
}
```

Reply:
ahh good idea.